salinity/salinity_front/views.py

```diff
@@ -8,26 +8,42 @@
 import logging
 import collections
 from salinity_front.models import CheckRedis
+import jsonpickle
+from time import time
+import threading
+
+server_con = CheckRedis("localhost")
+roles = {'app':['web', 'lb', 'php', 'app', 'util', 'queue', 'solr', 'es', 'node', 'nfs', 'sftp', 'rsyslog', 'mmonit'], 'ci': ['slave', 'ci', 'tarantula', 'awsutils']}
+no_stg = ['rsyslog', 'mmonit']
+envs = {'app':['qa', 'stg', 'prd'], 'ci':['ci']}
+
+def update_redis_context():
+    while(True):
+        context_dict = {}
+        for env_type, env_list in envs.iteritems():
+            for env in env_list:
+                for role in roles[env_type]:
+                    context_dict[role + "_" + env] = {'status':server_con.check_failed_role(role, env), 'role':role, 'env':env}
+        redis_context = jsonpickle.encode(context_dict)
+        server_con.write_context(redis_context)
```

Comment:
would it make sense to pull data from redis "once per time period + whatever it takes to pull from redis"?    otherwise you can 100% consume 1 CPU  with while(True) loop.  (i.e. add some sleep( time period) )